test/mlir/onnx/onnx_shape_inference.mlir

```diff
@@ -2188,6 +2188,64 @@ func @test_random_normal_static_f64() -> tensor<*xf32> {
   // CHECK: [[R0:%.+]] = "onnx.RandomNormal"() {dtype = 2 : si64, mean = 0.000000e+00 : f32, scale = 1.000000e+00 : f32, seed = 2.000000e+00 : f32, shape = [3, 4, 5]} : () -> tensor<3x4x5xf64>
 }
 
+//===----------------------------------------------------------------------===//
+
+// -----
+
+// Test RandomNormalLike static
+
+func @test_random_normal_like_static_f16(%arg0: tensor<1x1x28x28xf32>) -> tensor<*xf32> {
+  %0 = "onnx.RandomNormalLike"(%arg0) {dtype = 0 : si64, mean = 0.0 :f32, scale = 1.0 : f32, seed = 2.0 : f32} : (tensor<1x1x28x28xf32>) -> tensor<*xf32>
```

Comment:
`dtype = 0` means the operator will produce a result tensor with element type `f16`, therefore (maybe I am missing something) the output tensor of this operator should be `tensor<*xf16>` ?

Reply:
Since this is the user input code it doesn't matter. The type is f16 where it matters which is in the generated code. This test shows that even if the user supplies the wrong type to start with, shape inference will enforce the `dtype` type.